src/Cms/AppPlugins.php

```diff
@@ -713,17 +713,22 @@ public function nativeComponent(string $component): Closure|false
 	 */
 	public static function plugin(
 		string $name,
-		array $extends = null
+		array $extends = null,
+		array $info = [],
+		string|null $root = null
 	): PLugin|null {
 		if ($extends === null) {
 			return static::$plugins[$name] ?? null;
 		}
 
-		// get the correct root for the plugin
-		$extends['root'] ??= dirname(debug_backtrace()[0]['file']);
+		$plugin = new Plugin(
+			name:    $name,
+			extends: $extends,
+			info:    $info,
+			root:    $root ?? dirname(debug_backtrace()[0]['file']),
```

Comment:
We need to still support the old `$extends` way, otherwise it would be an immediate breaking change:

```suggestion
			// TODO: Remove fallback to $extends in v5
			root:    $root ?? $extends['root'] ?? dirname(debug_backtrace()[0]['file']),
```